cub/cub/thread/thread_operators.cuh

```diff
@@ -44,9 +44,11 @@ _CCCL_IMPLICIT_SYSTEM_HEADER
 #include <cub/util_cpp_dialect.cuh>
 #include <cub/util_type.cuh>
 
+_CCCL_SUPPRESS_DEPRECATED_PUSH
 #include <cuda/std/functional>
 #include <cuda/std/type_traits>
 #include <cuda/std/utility>
+_CCCL_SUPPRESS_DEPRECATED_POP
```

Comment:
I think this was only needed for functional.

Ideally we should fix the libcu++ headers to not use their own deprecated functionality so the headers won't throw warnings from just including them. Should we open an issue?

Reply:
Oh we have some unholy include things going on so I wanted to be sure to keep utility in it.

We arent really using deprecated features, they are just defined